### PR TITLE
add wrapper class for generic scraper

### PR DESCRIPTION
**Pull request by DengYiping**
This is a generic class, which include a builder.
This builder will help those people who work on scraper to insert data to loklak.
#751

### CODE REVIEW

File: src/org/loklak/data/DAOWrapper.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ "location_point": [3.058579854228782,50.63296878274201],
+ "location_radius": 0,
+ "user": {
+ "user_id": "youruserid_5678",
+ "name": "Mr. Bob",
+ }
+ }
+ ]
+ }
+ */
```
Comment (jig08) [inline src/org/loklak/data/DAOWrapper.java]: @DengYiping So as all scrapers have different formats, how should we go about pushing them into the index using the builder ? Does, everything go into "text" (considering the fact we have more fields than the Twitter scarper) ?

**Comment by DengYiping** [comment]
@jig08 Yes. Using markdown format, and insert it into text according to what Orbiter said.

**Comment by DengYiping** [comment]
@jig08  I will modify a little bit if you can describe the schema you are gonna use.

**Comment by jig08** [comment]
@DengYiping For instance, I want to push the data obtained using [http://localhost:9000/api/console.json?q=SELECT%20*%20FROM%20locationwisetime%20WHERE%20query=%27london%27;]http://localhost:9000/api/console.json?q=SELECT%20*%20FROM%20locationwisetime%20WHERE%20query=%27london%27;)

which is as :
```
{
  "data": [
    {
      "location": "Canada - Ontario - London",
      "time": "Sun 9:33 PM"
    },
    {
      "location": "United Kingdom - England - London",
      "time": "Canada - Ontario - London *"
    },
    {
      "location": "South Africa - East London",
      "time": "Mon 3:33 AM"
    },
    {
      "location": "United Kingdom - Northern Ireland - Londonderry",
      "time": "South Africa - East London"
    }
  ],
  "metadata": {"count": 4},
  "session": {"identity": {
    "type": "host",
    "name": "0:0:0:0:0:0:0:1",
    "anonymous": true
  }}
}
```

**Comment by DengYiping** [comment]
@jig08  I see here, there are 2 location. It won't fit into twitter schema, so the extra one should go to text field. I let me further generalize these message, do you have more examples of messages?

**Comment by jig08** [comment]
@DengYiping Sure. Shall list 'em here....

**Comment by jig08** [comment]
* RSS Reader: [http://loklak.org/api/rssreader.json?url=https://news.ycombinator.com/rss](http://loklak.org/api/rssreader.json?url=https://news.ycombinator.com/rss)

```
 {
      "link": "https://techcrunch.com/2016/08/01/salesforce-buys-word-processing-app-quip-for-750m/",
      "guid": "-729948106",
      "description": "<a href=\"https://news.ycombinator.com/item?id=12205855\">Comments<\/a>",
      "title": "Salesforce buys word processing app Quip for $750M",
      "uri": "https://techcrunch.com/2016/08/01/salesforce-buys-word-processing-app-quip-for-750m/",
      "pubDate": "Mon Aug 01 15:54:08 CDT 2016"
    }
```

* Github pROFILE sCRAPER: [http://loklak.org/api/console.json?q=SELECT%20*%20FROM%20githubProfile%20WHERE%20profile=%27mariobehling%27;](http://loklak.org/api/console.json?q=SELECT%20*%20FROM%20githubProfile%20WHERE%20profile=%27mariobehling%27;)

```
{
  "data": [{
    "joining_date": "2012-03-28T15:13:30Z",
    "gists_url": "https://api.github.com/users/mariobehling/gists",
    "repos_url": "https://api.github.com/users/mariobehling/repos",
    "user_name": "mariobehling",
    "following_url": "https://github.com/mariobehling/following",
    "bio": "",
    "starred_url": "https://github.com/stars/mariobehling",
    "followers_url": "https://github.com/mariobehling/followers",
    "subscriptions_url": "https://api.github.com/users/mariobehling/subscriptions",
    "received_events_url": "https://api.github.com/users/mariobehling/received_events",
    "full_name": "Mario Behling",
    "followers": "108 Followers",
    "avatar_url": "https://avatars3.githubusercontent.com/u/1583873?v=3&s=460",
    "starred": "4 Starred",
    "events_url": "https://api.github.com/users/mariobehling/events",
    "following": "39 Following",
    "organizations": [
      {
        "img_link": "https://avatars0.githubusercontent.com/u/865503?v=3&s=84",
        "link": "https://github.com/freifunk",
        "label": "freifunk",
        "img_Alt": "@freifunk"
      },
      {
        "img_link": "https://avatars0.githubusercontent.com/u/2297092?v=3&s=84",
        "link": "https://github.com/phimpme",
        "label": "phimpme",
        "img_Alt": "@phimpme"
      },
      {
        "img_link": "https://avatars3.githubusercontent.com/u/3916718?v=3&s=84",
        "link": "https://github.com/meilix",
        "label": "meilix",
        "img_Alt": "@meilix"
      },
      {
        "img_link": "https://avatars1.githubusercontent.com/u/4267331?v=3&s=84",
        "link": "https://github.com/oert",
        "label": "oert",
        "img_Alt": "@oert"
      },
      {
        "img_link": "https://avatars3.githubusercontent.com/u/6295529?v=3&s=84",
        "link": "https://github.com/fossasia",
        "label": "fossasia",
        "img_Alt": "@fossasia"
      },
      {
        "img_link": "https://avatars2.githubusercontent.com/u/7567896?v=3&s=84",
        "link": "https://github.com/fashiontec",
        "label": "fashiontec",
        "img_Alt": "@fashiontec"
      },
      {
        "img_link": "https://avatars2.githubusercontent.com/u/7839834?v=3&s=84",
        "link": "https://github.com/MeshCon",
        "label": "MeshCon",
        "img_Alt": "@MeshCon"
      },
      {
        "img_link": "https://avatars3.githubusercontent.com/u/7956151?v=3&s=84",
        "link": "https://github.com/OpenTechSummit",
        "label": "OpenTechSummit",
        "img_Alt": "@OpenTechSummit"
      },
      {
        "img_link": "https://avatars3.githubusercontent.com/u/11370631?v=3&s=84",
        "link": "https://github.com/loklak",
        "label": "loklak",
        "img_Alt": "@loklak"
      }
    ],
    "home_location": "",
    "works_for": "",
    "special_link": "http://mariobehling.de",
    "email": "",
    "atom_feed_link": "https://github.com/mariobehling.atom"
  }],
  "metadata": {"count": 1},
  "session": {"identity": {
    "type": "host",
    "name": "10.67.93.57",
    "anonymous": true
  }}
}
```

* Wordpress Crawler: [http://loklak.org/api/console.json?q=SELECT%20*%20FROM%20wordpress%20WHERE%20url=%27https://jigyasagrover.wordpress.com/%27;](http://loklak.org/api/console.json?q=SELECT%20*%20FROM%20wordpress%20WHERE%20url=%27https://jigyasagrover.wordpress.com/%27;)

```
{
  "data": [{
    "blog_url": "https://jigyasagrover.wordpress.com/",
    "title": "A niche for my kaleidoscopic explorations :)",
    "content": "Check out this space for posts spanning colossal themes ! #pharo #smalltalk #morphic #spec #java #android #c #c++ #git #php #asp.net #python Catch up with updates on Google Summer of Code project apprises, Google Code-In mentoring experiences, Conference reports, Google Developers Group events, Women Who Code MeetUps , Google Women Techmakers Get Togethers , contributions to organizations like FOSSASIA and Square Bracket Associates, sundry Startup experiences, Enactus undertakings, Social Welfare escapades and many more \u2026. Keep the feedback pouring in \u2026 Follow/like/comment for more. Post any query, Would be happy to help😀 Go to: RECENT POSTS   Share this: Twitter Facebook Google Like this: Like Loading..."
  }],
  "metadata": {"count": 1},
  "session": {"identity": {
    "type": "host",
    "name": "10.67.93.57",
    "anonymous": true
  }}
}
```

**Comment by jig08** [comment]
@DengYiping These are just a few examples ....

**Comment by DengYiping** [comment]
@jig08 OK, let me find their common point.

**Comment by DengYiping** [comment]
@jig08 This version should satisfy the majority of your need. If you need add more information, use `addText(String text_)` method.(all method start with `add` can be used multiple times, and with `set` can only use one time.). Finally use `persist()` method to insert the data into elasticsearch.

**Comment by jig08** [comment]
@DengYiping Thanks for the elaboration. Will try this out for pushing data in the index...

**Comment by sudheesh001** [comment]
@DengYiping It'd be great if it's possible for pushing any JSON format over to an index.

**Comment by jig08** [comment]
@DengYiping A few `codacy` checks are unsuccessful, kindly look through them ...

**Comment by DengYiping** [comment]
@jig08 The codacy is just about my code style. It is not a problem if you click into it and check. I always ignore those warnings.

Travis-CI is passed, and that is what all matters.
By the way, I've just write a interface, feel free to check out.

**Comment by jig08** [comment]
@DengYiping I guess you simple need to resolve the issue enlisted here 
Link: [codacy.com/app/sudheesh1995/loklak_server/pullRequest?prid=302704](https://www.codacy.com/app/sudheesh1995/loklak_server/pullRequest?prid=302704)

**Comment by jig08** [comment]
@DengYiping Agreed. I thought maybe we should adhere to these code coverage and quality integrations.

Yes, testing the interface currently.

**Comment by DengYiping** [comment]
@jig08 Code coverage is fixed.

**Pull request merged**